app/code/Mykhailok/SupportChat/Model/MessageUserDataProvider.php

```diff
@@ -3,58 +3,85 @@
 
 namespace Mykhailok\SupportChat\Model;
 
-/**
- * Class MessageUserDataProvider
- * Provide method getChatMessageWithUserData();
- */
 class MessageUserDataProvider
 {
-    /**
-     * @var \Mykhailok\SupportChat\Model\ChatMessageFactory
-     */
-    private $chatMessageFactory;
-
-    /**
-     * @var \Magento\Store\Model\StoreManagerInterface
-     */
-    private $storeManager;
+    protected ?\Mykhailok\SupportChat\Model\Chat $chatModel;
```

Comment:
@mykhailok-dv 
1) This looks like a private property
2) `?` should not be needed. Any property can be NULL before it is accessed. Here is a code example:
```php
class Foo
{
    private ?FooBarBaz $bar;

    public function getBar()
    {
        if (!isset($this->bar)) {
            echo "Not set\n";
        } else {
            echo "Set\n";
        }
    }
}

$foo = new Foo();
$foo->getBar();
```
As you can see, PHP even does not fail despite that the class `FooBarBaz` does not exist - it does not load it because the property is not set and there is no need to check the property type yet.